hw3/01-routing.js

```diff
@@ -36,11 +54,43 @@ const server = http.createServer((req, res) => {
 
   if (req.url === '/') {
     let routeResults = getRoutes();
-
     res.writeHead(200, { 'Content-Type': 'text/html' });
     res.write(`<h1>Exercise 01</h1>`);
     res.write(`<ul> ${routeResults} </ul>`);
     res.end();
+  } else if (req.url === '/welcome') {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Hello there! Welcome!</h1>`);
+    res.end();
+  } else if (req.url === '/redirect') {
+    res.redirect(302, `http://localhost:${port}/redirected`);
+    res.end();
+  } else if (req.url === '/redirected') {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Redirected</h1>`);
+    res.end();
+  } else if (req.url === '/cache') {
+    res.setHeader('Cache-control', 'public, max-age=86400');
+    res.setHeader('content-type', 'text/html');
+    res.write(`<h1>this resource was cached</h1>`);
+    res.end();
+  } else if (req.url === '/cookie') {
+    res.setHeader('Set-Cookie', cookie.serialize('hello', 'world'));
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>cookies... yummm</h1>`);
+    res.end();
+  } else if (req.url === '/check-cookies') {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    const cookies = parseCookies(req);
+    if (cookies.hasOwnProperty('hello')) {
+      res.write(`<h1>yes</h1>`);
+    } else {
+      res.write(`<h1>no</h1>`);
+    }
+    res.end();
+  } else {
+    res.writeHead(404, { 'Content-Type': 'text/html' });
+    res.write(`<h1>404 Page Not Found</h1>`);
   }
```

Comment:
Good job overall:
- Strings: Good, you used '' for everything.
- Interpolation: Good your string interpolations are easy to read and follow.
- Naming Conventions: Good, everything seems to be camelCase